modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java

```diff
@@ -695,16 +696,36 @@ private int _dynamicSearchCount(
 		return availableClassTypeIds;
 	}
 
-	private long[] _getCombinedSegmentsEntryIds(long[] segmentEntryIds) {
+	private long[] _getCombinedSegmentsEntryIds(
+		AssetListEntry assetListEntry, long[] segmentEntryIds) {
+
 		if ((segmentEntryIds.length > 1) &&
 			ArrayUtil.contains(
 				segmentEntryIds, SegmentsEntryConstants.ID_DEFAULT)) {
 
-			return ArrayUtil.remove(
+			segmentEntryIds = ArrayUtil.remove(
 				segmentEntryIds, SegmentsEntryConstants.ID_DEFAULT);
 		}
 
-		return segmentEntryIds;
+		LongStream longStream = Arrays.stream(segmentEntryIds);
+
+		Stream<AssetListEntrySegmentsEntryRel>
+			assetListEntrySegmentsEntryRelStream = longStream.mapToObj(
+				segmentsEntryId ->
+					_assetListEntrySegmentsEntryRelLocalService.
+						fetchAssetListEntrySegmentsEntryRel(
+							assetListEntry.getAssetListEntryId(),
+							segmentsEntryId));
+
+		return assetListEntrySegmentsEntryRelStream.filter(
```

Comment:
maybe we can avoid declare assetListEntrySegmentsEntryRelStream, or SF force to declare it?

Reply:
Apparently SF does not comply about it, just reworded to make it simple